sirepo/package_data/static/js/srw.js

```diff
@@ -2194,6 +2189,10 @@ SIREPO.app.directive('simulationStatusPanel', function(appState, beamlineService
                 srwService.computeModel($scope.model),
                 handleStatus
             );
+
+            $scope.simState.notRunningMessage = function() {
```

Comment:
This seems strange as there's no indication here that the simulation is or is not running. This is just displaying stateAsText with "Simulation " in front of it.

Reply:
Can you say that a different way? I don't understand the issue. 

If the simulation isn't running it will say something like `Simulation completed`. This code is [only called](https://github.com/radiasoft/sirepo/blob/6d285a3f4e33e9f4871aa63417f5451f8947d4a0/sirepo/package_data/static/js/sirepo-components.js#L1462) when a simulation isn't running

Reply:
```js
   self.simState.notRunningMessage = function() {
        return 'Simulation ' + self.simState.stateAsText();
    };
```

is called here:

```js
             var m = 'notRunningMessage';
                var res = $scope.simState[m] && $scope.simState[m]();
                if (res) {
                    return res;
                }
                res = 'Simulation ' + $scope.simState.stateAsText();
```

The concept 'Simulation ' + stateAsText can be reused in all cases. They are all the same format.

The problem is `simState.getFrameCount()` needs to be called in the case that it is running or has run. That's a separate concept but related to simState. Therefore, I propose a single function on simState that does it all.

Reply:
I've abstracted `'Simulation' + stateAsText()`. https://github.com/radiasoft/sirepo/pull/2690/commits/52c29708d04620911b0275d8c8fef7b493ebc23e . I have a feeling you want something more but I'm not sure.

`simState.getFrameCount()` needs to be called when we **care** about frames. Ex in srw we care about particles not frames so we have a different notRunningMessage that doesn't display info about frames.

Reply:
@robnagler https://github.com/radiasoft/sirepo/pull/2690/commits/71a952b86573f08ba59907d977720a7a4aeb0a7e